File: schemaorg/prependaction.go
package schemaorg

import "encoding/json"

// PrependAction see : https://schema.org/PrependAction
type PrependAction struct {

typeContext

InsertAction

// ToLocation see : /toLocation
// A sub property of location. The final location of the object or the agent after the action.
ToLocation *Place `json:"toLocation"`

}

func (v *PrependAction) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "PrependAction"

	return json.Marshal(v)
}

File: schemaorgphp/classes/prepend_action.php
<?php

class PrependAction extends InsertAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'PrependAction';
	
	/**
	 * A sub property of location. The final location of the object or the agent after the action.
	 * see : https://schema.org/toLocation
	 * @var \Place|\Place[]
	 */
	public var $to_location;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'PrependAction'
		);
		
		$serialized = so_json_serialize( $this->to_location );
		if ( ! empty( $serialized ) ) {
			$out['toLocation'] = $serialized;
		}
		
		return $out;
	}
}
